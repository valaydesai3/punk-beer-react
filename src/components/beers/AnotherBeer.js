import React, { useContext } from 'react';
import PunkBeerContext from '../../context/punkBeerContext';

const AnotherBeer = () => {
  const punkBeerContext = useContext(PunkBeerContext);
  return (
    <div data-testid="anotherBeerBtn">
      <button
        className="btn"
        onClick={() => punkBeerContext.getRandomBeer(false)}
      >
        Another Beer
      </button>
    </div>
  );
};

export default AnotherBeer;
