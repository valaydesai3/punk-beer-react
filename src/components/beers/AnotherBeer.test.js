// __tests__/fetch.test.js
import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import 'jest-dom/extend-expect';
import PunkBeerContext from '../../context/punkBeerContext';
import AnotherBeer from './AnotherBeer';

afterEach(cleanup);

test('renders another beer button', async () => {
  let getRandomBeer = jest.fn();

  const { getByText, getByTestId } = render(
    <PunkBeerContext.Provider
      value={{
        getRandomBeer
      }}
    >
      <AnotherBeer />
    </PunkBeerContext.Provider>
  );
  expect(getByTestId('anotherBeerBtn')).toHaveTextContent('Another Beer');

  fireEvent.click(getByText('Another Beer'));
  expect(getRandomBeer).toHaveBeenCalled();
  expect(getByTestId('anotherBeerBtn')).toMatchSnapshot();
});
