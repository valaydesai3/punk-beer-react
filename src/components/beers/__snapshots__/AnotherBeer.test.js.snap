// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders another beer button 1`] = `
<div
  data-testid="anotherBeerBtn"
>
  <button
    class="btn"
  >
    Another Beer
  </button>
</div>
`;
